modules/gapi/src/streaming/onevpl/accelerators/accel_policy_cpu.cpp

```diff
@@ -0,0 +1,231 @@
+#endif
+
+    if (!preallocated_pool_memory_ptr) {
+        throw std::runtime_error("VPLCPUAccelerationPolicy::create_surface_pool - failed: not enough memory."
+                                 "Requested surface count: " + std::to_string(pool_size) +
+                                 ", surface bytes: " + std::to_string(surface_size_bytes));
+    }
+
+    // fill pool with surfaces
+    std::shared_ptr<void> workspace_mem_owner (preallocated_pool_memory_ptr, [] (void *ptr){
+        GAPI_LOG_INFO(nullptr, "Free workspace memory: " << ptr);
+#ifdef _WIN32
+        _aligned_free(ptr);
+        GAPI_LOG_INFO(nullptr, "Released workspace memory: " << ptr);
+        ptr = nullptr;
+#else
+        abort(); //not implemented
+#endif
+
+        });
+    size_t i = 0;
+    try {
+        for (; i < pool_size; i++) {
+            size_t preallocated_mem_offset = static_cast<size_t>(i) * surface_size_bytes;
+
+            surface_ptr_t surf_ptr = creator(workspace_mem_owner,
+                                             preallocated_mem_offset,
+                                             preallocated_raw_bytes);
+            pool.push_back(std::move(surf_ptr));
+        }
+    } catch (const std::exception& ex) {
+        throw std::runtime_error(std::string("VPLCPUAccelerationPolicy::create_surface_pool - ") +
+                                 "cannot construct surface index: " + std::to_string(i) + ", error: " +
+                                 ex.what() +
+                                 "Requested surface count: " + std::to_string(pool_size) +
+                                 ", surface bytes: " + std::to_string(surface_size_bytes));
+    }
+
+    // remember pool by key
+    GAPI_LOG_INFO(nullptr, "New pool allocated, key: " << preallocated_pool_memory_ptr <<
+                           ", surface count: " << pool.size() <<
+                           ", surface size bytes: " << surface_size_bytes);
+    try {
+        if (!pool_table.emplace(preallocated_pool_memory_ptr, std::move(pool)).second) {
+            throw std::runtime_error(std::string("VPLCPUAccelerationPolicy::create_surface_pool - ") +
+                                     "cannot insert pool, table size: " + std::to_string(pool_table.size()));
+        }
+    } catch (const std::exception&) {
+        throw;
+    }
+
+    return preallocated_pool_memory_ptr;
+}
+
+VPLCPUAccelerationPolicy::surface_weak_ptr_t VPLCPUAccelerationPolicy::get_free_surface(pool_key_t key) {
+    auto pool_it = pool_table.find(key);
+    if (pool_it == pool_table.end()) {
+        std::stringstream ss;
+        ss << "key is not found: " << key << ", table size: " << pool_table.size();
+        const std::string& str = ss.str();
+        GAPI_LOG_WARNING(nullptr, str);
+        throw std::runtime_error(std::string(__FUNCTION__) + " - " + str);
+    }
+
+    pool_t& requested_pool = pool_it->second;
+#ifdef TEST_PERF
```

Comment:
Didn't quite get it

Reply:
it was implemented for perf test reasons, because naïve vector-based implementation was suspicious in worst performance but existing single_roi_example wouldn't have show any difference for infer-scenario. Further PRs with performance improvement can prove degradation after introducing new scope of performance test for decode-only case without infer. This macro will be removed and only new cached poll will be remained in next PRs